### PR TITLE
Get hearers in view faster with 0 range

### PR DESCRIPTION
**Pull request by TiviPlus**
## About The Pull Request
0 Range:
image
8 and 5 range are a teeny bit faster in the profiler but not too much
https://github.com/tgstation/TerraGov-Marine-Corps/pull/5436/files
## Changelog
:cl:
code: Made get_hearers_in_view faster
/:cl:

### CODE REVIEW

File: code/__HELPERS/game.dm [hunk 1]
```diff
@@ -217,31 +217,28 @@
...
-	while(i < length(processing_list)) // recursive_hear_check inlined here
-		var/atom/A = processing_list[++i]
+	while(i < length(.)) // recursive_hear_check inlined here
+		var/atom/A = .[++i]
 		if(A.flags_1 & HEAR_1)
-			. += A
-			SEND_SIGNAL(A, COMSIG_ATOM_HEARER_IN_VIEW, processing_list, .)
-		processing_list += A.contents
+			SEND_SIGNAL(A, COMSIG_ATOM_HEARER_IN_VIEW, .)
+		. += A.contents
```
Comment (LemonInTheDark) [inline code/__HELPERS/game.dm:241]: How hot is this proc exactly? is it worth this level of micro op?

Comment (TiviPlus) [reply code/__HELPERS/game.dm:241]: Notable uses
Say()
visible_message
Flashes & flashbangs
Instruments
Mech speech handling again

Comment (TiviPlus) [reply code/__HELPERS/game.dm:241]: So yes pretty often

File: code/__HELPERS/game.dm [hunk 2]
```diff
@@ -217,31 +217,28 @@
...
 	var/i = 0
-	while(i < length(processing_list)) // recursive_hear_check inlined here
-		var/atom/A = processing_list[++i]
+	while(i < length(.)) // recursive_hear_check inlined here
+		var/atom/A = .[++i]
 		if(A.flags_1 & HEAR_1)
-			. += A
-			SEND_SIGNAL(A, COMSIG_ATOM_HEARER_IN_VIEW, processing_list, .)
-		processing_list += A.contents
+			SEND_SIGNAL(A, COMSIG_ATOM_HEARER_IN_VIEW, .)
```
Comment (Ghommie) [inline code/__HELPERS/game.dm:240]:
```suggestion
			SEND_SIGNAL(A, COMSIG_ATOM_HEARER_IN_VIEW, .)
		else
			. -= A
			i--
```
`get_hearers_in_view` is supposed to return a list of movables with HEAR_1 enabled and no more, comsig aside.

File: code/__HELPERS/game.dm [hunk 3]
```diff
@@ -217,31 +217,31 @@
...
+		else
+			. -= A
```
Comment (LemonInTheDark) [inline code/__HELPERS/game.dm:242]: This strikes me as an operation that would make the proc slower rather then faster, could you detail how you did the profiling in greater detail?

Comment (Rohesie) [reply code/__HELPERS/game.dm:242]: Wouldn't it be better to have a second auxiliary list that is `.Cut()` at the end of each loop?

**Pull request closed without merging**